x-pack/plugins/security_solution/server/search_strategy/event_filters_fields/index.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { from } from 'rxjs';
+import type {
+  DataViewsServerPluginStart,
+  ISearchStrategy,
+  SearchStrategyDependencies,
+} from '@kbn/data-plugin/server';
+
+import { requestIndexFieldSearch } from '@kbn/timelines-plugin/server/search_strategy/index_fields';
+
+import { eventsIndexPattern } from '../../../common/endpoint/constants';
+import type {
+  BeatFields,
+  IndexFieldsStrategyRequest,
+  IndexFieldsStrategyResponse,
+} from '../../../common/search_strategy';
+import type { EndpointAppContextService } from '../../endpoint/endpoint_app_context_services';
+
+/**
+ * EventFiltersFieldProvider mimics indexField provider from timeline plugin: x-pack/plugins/timelines/server/search_strategy/index_fields/index.ts
+ * but it uses ES internalUser instead to avoid adding extra index privileges for users with event filters permissions.
+ * It is used to retrieve index patterns for event filters form.
+ */
+export const eventFiltersFieldsProvider = (
+  context: EndpointAppContextService,
+  indexPatterns: DataViewsServerPluginStart
+): ISearchStrategy<IndexFieldsStrategyRequest<'indices'>, IndexFieldsStrategyResponse> => {
+  // require the fields once we actually need them, rather than ahead of time, and pass
+  // them to createFieldItem to reduce the amount of work done as much as possible
+  const beatFields: BeatFields =
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    require('@kbn/timelines-plugin/server/utils/beat_schema/fields').fieldsBeat;
+
+  return {
+    search: (request, _, deps) =>
+      from(requestEventFiltersFieldsSearch(context, request, deps, beatFields, indexPatterns)),
+  };
+};
+
+export const requestEventFiltersFieldsSearch = async (
+  context: EndpointAppContextService,
+  request: IndexFieldsStrategyRequest<'indices'>,
+  deps: SearchStrategyDependencies,
+  beatFields: BeatFields,
+  indexPatterns: DataViewsServerPluginStart
+): Promise<IndexFieldsStrategyResponse> => {
+  const { canWriteEventFilters } = await context.getEndpointAuthz(deps.request);
+
+  if (!canWriteEventFilters) {
+    throw new Error('Endpoint authz error');
```

Comment:
Maybe use `EndpointAuthorizationError` ([here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/endpoint/errors.ts#L26-L30)) instead? which should cause the API to return a `403`

Reply:
https://github.com/elastic/kibana/pull/145883/commits/28a760f1c1d9888a6cfbb9d8b40b36f7e8f8b27f